docs/sdk/src/reference_docs/wasm_meta_protocol.rs

```diff
@@ -90,24 +94,66 @@
 //!
 //! In fact, [`sp_core::storage::well_known_keys`] are the only state keys that the node side is
 //! aware of. The rest of the state, including what logic the runtime has, what balance each user
-//! has and such are all only comprehensible to the runtime.
+//! has and such, are all only comprehensible to the runtime.
 #![doc = simple_mermaid::mermaid!("../../../mermaid/state.mmd")]
 //!
 //! In the above diagram, all of the state keys and values are opaque bytes to the node. The node
-//! does not know what they mean, and it does not now what is the type of the corresponding value
+//! does not know what they mean, and it does not know what is the type of the corresponding value
 //! (e.g. if it is a number of a vector). Contrary, the runtime knows both the meaning of their
 //! keys, and the type of the values.
 //!
-//! This opaque-ness is the fundamental reason why Substrate-based chains can fork-less-ly upgrade:
-//! because the node side code is kept oblivious to all of the details of the state transition
-//! function. Therefore, the state transition function can freely upgrade without the node needing
-//! to know.
+//! This opaque-ness is the fundamental reason why Substrate-based chains can fork-less-ly
+//! upgrade: because the node side code is kept oblivious to all of the details of the state
+//! transition function. Therefore, the state transition function can freely upgrade without the
+//! node needing to know.
 //!
 //! ## Native Runtime
 //!
-//! TODO
+//! Historically, the node software also kept a native copy of the runtime at the time of
+//! compilation within it. This used to be called the "Native Runtime". The main purpose of the
+//! native runtime used to be leveraging the faster execution time and better debugging
+//! infrastructure of native code. However, neither of the two arguments strongly hold and the
```

Comment:
nit: Is really debugging wasm runtime as easy as native one?
I would maybe mention the huge  maintenance burden of native execution path, and possibly nondeterminism of native vs wasm e.g. when compile with different compilers. Not sure if we want to get into this here.